### PR TITLE
Revisit macos (#2)

### PR DESCRIPTION
**Pull request by mhaley-tignis**
chore: add macos arm build

---------

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -4,8 +4,8 @@ build-backend = "setuptools.build_meta"
 
 [tool.cibuildwheel]
 build-frontend = "build"
-build = "cp3*"
-skip = ["*-win32", "*-manylinux_i686", "*-musllinux_*"]
+build = "cp38*"
```
Comment (wbarnha) [inline pyproject.toml]: Pretty sure this results in wheels being built for only Python 3.8. Change this to `cp3*` and we should be good.

File: .github/workflows/dist.yml [hunk 2]
```diff
@@ -2,18 +2,20 @@
...
 
 on:
-  release:
-    types: [created]
+  pull_request:
+  push:
     branches:
       - 'master'
+      - 'force_ci/all/**'     # For development, forcing all workflows to run.
+      - 'force_ci/debian/**'  # For debugging and/or only forcing this workflow.
```
Comment (mhaley-tignis) [inline .github/workflows/dist.yml]: @wbarnha do we need to revert these changes too?

Comment (wbarnha) [reply .github/workflows/dist.yml]: Yes, please revert

**Review by wbarnha — APPROVED** [review]
